src/OpenLoco/Windows/ScenarioSelect.cpp

```diff
@@ -0,0 +1,507 @@
+
+            // Description
+            auto str = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+            strncpy(str, scenarioInfo->description, std::size(scenarioInfo->description));
+
+            auto args = FormatArguments();
+            args.push(StringIds::buffer_2039);
+            y = Gfx::drawString_495224(*dpi, x, y, 170, Colour::black, StringIds::black_stringid, &args);
+
+            // Challenge header
+            y += 5;
+            auto origin = Gfx::point_t(x, y);
+            Gfx::drawString_494B3F(*dpi, &origin, Colour::black, StringIds::challenge_label, nullptr);
+
+            // Challenge text
+            str = const_cast<char*>(StringManager::getString(StringIds::buffer_1250));
+            strncpy(str, scenarioInfo->objective, std::size(scenarioInfo->objective));
+
+            y += 10;
+            args = FormatArguments();
+            args.push(StringIds::buffer_1250);
+            y = Gfx::drawString_495224(*dpi, x, y, 170, Colour::black, StringIds::challenge_value, &args);
+
+            // Start year
+            y += 5;
+            args = FormatArguments();
+            args.push(scenarioInfo->startYear);
+            Gfx::drawString_494B3F(*dpi, x, y, Colour::black, StringIds::challenge_start_date, &args);
+
+            // Competing companies
+            y += 10;
+            args = FormatArguments();
+            args.push<uint16_t>(scenarioInfo->numCompetingCompanies);
+            string_id competitionStringId = scenarioInfo->numCompetingCompanies == 0 ? StringIds::challenge_competing_companies_none : StringIds::challenge_competing_companies_up_to;
+            y = Gfx::drawString_495224(*dpi, x, y, 170, Colour::black, competitionStringId, &args);
+
+            if (scenarioInfo->numCompetingCompanies == 0 || scenarioInfo->competingCompanyDelay == 0)
+                return;
+
+            // Delayed start for competing companies
+            args = FormatArguments();
+            args.push<uint16_t>(scenarioInfo->competingCompanyDelay);
+            competitionStringId = scenarioInfo->numCompetingCompanies == 1 ? StringIds::competition_not_starting_for_month : StringIds::competition_not_starting_for_months;
+            Gfx::drawString_494B3F(*dpi, x, y, Colour::black, competitionStringId, &args);
+        }
+    }
+
+    // 0x00443D02
+    static void drawScroll(window* self, Gfx::drawpixelinfo_t* const dpi, uint32_t)
+    {
+        auto colour = Colour::getShade(self->colours[1], 4);
+        Gfx::clearSingle(*dpi, colour);
+
+        using namespace ScenarioManager;
+        auto scenarioCount = getScenarioCountByCategory(self->current_tab);
+
+        int16_t y = 0;
+        for (auto i = 0; i < scenarioCount; i++)
+        {
+            auto* scenarioInfo = getNthScenarioFromCategory(self->current_tab, i);
+            if (scenarioInfo == nullptr)
+                continue;
+
+            if (y + rowHeight < dpi->y || y > dpi->y + dpi->height)
```

Comment:
This i suspect is the source of the flickering bug. You can't use the dpi x, y, height and width as it can be split up by the renderer

Reply:
Never mind above was correct. I found the issue though you are missing a += rowHeight on the y.